packages/block-library/src/button/edit.js

```diff
@@ -92,19 +102,34 @@ function ButtonEdit( props ) {
 
 	const TagName = tagName || 'a';
 
-	function onToggleOpenInNewTab( value ) {
-		const newLinkTarget = value ? '_blank' : undefined;
+	function updateLinkAttributes( newUrl, opensInNewWindow, nofollow ) {
```

Comment:
I feel this function is sufficiently complex to warrant a small set of unit tests. These can live alongside the block in the `test` folder in a file called `update-link-attributes.js`.

The tests should validate that the new value is correctly computed given various values that could be pased to it.

I'm keen that we test the regex.

Reply:
Good idea! I'll give it a try

Reply:
@getdave, since `updateLinkAttributes` is defined inside the functional component and can’t be accessed from outside the component. Therefore my approach would be moving the logic into a separate function in `/utils.js`, so we can test that the attributes generation is correct. Please let me know your thoughts. Thanks 🙇 

You can run the test by `npm run test:unit -- packages/block-library/src/button`